weatherApp/App/AppDelegate.swift

```diff
@@ -11,16 +11,19 @@ import UIKit
 @UIApplicationMain
 class AppDelegate: UIResponder, UIApplicationDelegate {
     
+    private let appDependencies = AppDependencies()
     var window: UIWindow?
-    
+    var navigationController: UINavigationController!
+
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+        
         window = UIWindow(frame: UIScreen.main.bounds)
+        navigationController = UINavigationController()
```

Comment:
You can initialize navigation controller iniside init of NavigationService and set this as:
```
private weak var navigationController: UINavigationController?
```

Reply:
If I initialise navigationController as weak var in NavigationServices constructor, I do not make any strong references to the navCon and therefore it's immediately released from memory. Solution Igor came up with is to make a strong reference of navigationController in AppDelegate which is forwarded to NavigationService where is used with a weak reference